core/types/deposit_request.go

```diff
@@ -70,37 +68,42 @@ type DepositRequestJson struct {
 
 func (d *DepositRequest) RequestType() byte { return DepositRequestType }
 func (d *DepositRequest) EncodeRLP(w io.Writer) (err error) {
```

Comment:
`EncodeRLP`/`DecodeRLP` are no longer accurate because it's not RLP anymore.

Reply:
Yes, that's right. It was just following suite till now anyways and never was the full RLP encoding. This will be removed in subsequent PRs

Reply:
Furthermore, the RLP encoding of the flat version would be almost what it is here, and would still work with Engine API and block storage, hence leaving it for this PR.